tests/data/intents/litexa/main.litexa

```diff
@@ -18,10 +18,55 @@ launch
     say "heard upper"
   when "lower case utterance"
     say "heard lower"
+  when OtherIntentName
+    or "other intent name"
+    -> someState
+
+someState
+  say "in some state"
+
+  when AMAZON.NoIntent
+    or OtherIntentName
+    say "got {context.event.request.intent.name}"
+
+  when "handler is utterance"
+    or lowercase
+    say "hello"
```

Comment:
Would keep these handlers consistent for what you're testing: You can simply have each of them say the expected intent name and test for that (i.e. HANDLER_IS_UTTERANCE here).